demo_nodes_py/demo_nodes_py/events/matched_event_detect.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 Sony Group Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rclpy
+from rclpy.event_handler import PublisherEventCallbacks
+from rclpy.event_handler import QoSPublisherMatchedInfo
+from rclpy.event_handler import QoSSubscriptionMatchedInfo
+from rclpy.event_handler import SubscriptionEventCallbacks
+from rclpy.executors import ExternalShutdownException
+from rclpy.node import Node
+
+from std_msgs.msg import String
+
+'''
+ This demo program shows detected matched event.
+ Matched event occur while connection between publisher and subscription is done.
+ Run this demo at one console by below command
+ $ ros2 run demo_nodes_py matched_event_detect
+ It will create 2 topics.
+ - 'pub_matched_event_detect' for detecting matched event of publisher
+ - 'sub_matched_event_detect' for detecting matched event of subscription
+
+ For checking matched event of publisher
+ On another console, run below command
+ $ ros2 topic echo /pub_matched_event_detect
+ You can run above command many times on different consoles to check the number of connected
+ subscription by the output of demo program.
+
+ For checking matched event of subscription
+ On another console, run below command
+ ros2 topic pub -r 1 /sub_matched_event_detect std_msgs/String "{data: '123'}"
+ You can run above command many times on different consoles to check the number of connected
+ publisher by the output of demo program.
+'''
+
+
+class MatchedEventDetectNode(Node):
```

Comment:
The behavior for the `matched event detect` in `demo_nodes_cpp` is different from `demo_nodes_py`, which might seem not good.

Reply:
Okay. I use the same logic as demo_nodes_cpp.